spec/models/user_spec.rb

```diff
@@ -0,0 +1,29 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  it { should have_valid(:first_name).when('Jon', 'Sam')}
+  it { should_not have_valid(:first_name).when(nil, '')}
+
+  it { should have_valid(:last_name).when('Smith', 'Jackson')}
+  it { should_not have_valid(:last_name).when(nil, '')}
+
+  it { should have_valid(:email).when('user@example.com', 'other@email.com')}
+  it { should_not have_valid(:email).when(nil, '', 'urser', 'users@')}
+
+  it { should have_valid(:user_name).when('Smith', 'Jackson')}
+  it { should_not have_valid(:user_name).when(nil, '')}
+
+  it { should have_valid(:state).when('MA', 'VT')}
+  it { should_not have_valid(:state).when(nil, '')}
+
+  it 'has a matching password confirmation for the password' do
+    user = User.new
+    user.password = 'password'
+    user.password_confirmation = 'whatever'
+
+    expect(user).to_not be_valid
+    expect(user.errors[:password_confirmation]).to_not be_blank
+  end
+
```

Comment:
Really nice model test! Don't forget to come back here and update it as you add associations to the `user` table. So, for instance, if a user `has_many` `reviews` in the future, you'll want to come back to this model spec and test for that new association! 

(Not that you should be doing this right now, just a heads up for the future)